invenio_app_ils/circulation/indexer.py

```diff
@@ -86,19 +87,24 @@ def index_extra_fields_for_loan(loan_dict):
     details.
     """
     document_class = current_app_ils.document_record_cls
-    document_record = document_class.get_record_by_pid(
-        loan_dict.get("document_pid")
-    )
+    try:
+        document_record = document_class.get_record_by_pid(
+            loan_dict["document_pid"]
+        )
+    except PIDDeletedError:
+        # Document might have been deleted while reindexing asyncronously.
+        return
+
     document = document_record.replace_refs()
 
     items_available_for_loan_count = document.get("circulation", {}).get(
-        "available_items_for_loan_count"
+        "available_items_for_loan_count", 0
```

Comment:
We should either include a default value or use [] if we always expect to get the value.
@kprzerwa I'm not sure if we always expect the value and, or, if the default value is needed, what should be it.